### PR TITLE
PIMS-1447 Updating financial data structure

### PR DESCRIPTION
**Pull request by Fosol**
Database financial data structure updated to the following;

```js
// Parcel
Parcel.evaluations[{ parcelId: 1, date: "2020/01/01", key: "Assessed", value: 324334.34 }, { parcelId: 1, date: "2020/01/01", key: "Appraised", value: 324334.34 }]
Parcel.fiscals[{ parcelId: 1, fiscalYear: 2020, key: "NetBook", value: 2343.34}, { parcelId: 1, fiscalYear: 2020, key: "Estimated", value: 2343.34}, ]

// Building
Building.evaluations[{ buildingId: 1, date: "2020/01/01", key: "Assessed", value: 324334.34 }, { buildingId: 1, date: "2020/01/01", key: "Appraised", value: 324334.34 }]
Building.fiscals[{ buildingId: 1, fiscalYear: 2020, key: "NetBook", value: 2343.34}, { buildingId: 1, fiscalYear: 2020, key: "Estimated", value: 2343.34}, ]
```

Note that **Evaluations** currently have two **Keys** ("Assessed", "Appraised").  These values are specific to the date they were performed.

Note that **Fiscals** currently have two **Keys** ("NetBook", "Estimated").  These values are specific to a fiscal year they are related to.

This will allow us to add additional financial values in the future without changing the database structure simply by updating the related *enum*.  The downside is that it is additional effort to fetch and find the values you want for any given **key**, **date** and **fiscal year**.

This should be the latest DB Refresh required (unless we find bugs).

### CODE REVIEW

File: backend/api/Areas/Admin/Mapping/Parcel/BuildingEvaluationMap.cs [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
 namespace Pims.Api.Areas.Admin.Mapping.Parcel
 {
     public class BuildingEvaluationMap : IRegister
     {
-
         public void Register(TypeAdapterConfig config)
         {
             config.NewConfig<Entity.BuildingEvaluation, Model.BuildingEvaluationModel>()
                 .IgnoreNonMapped(true)
+                .EnumMappingStrategy(EnumMappingStrategy.ByName)
```
Comment (Fosol) [inline backend/api/Areas/Admin/Mapping/Parcel/BuildingEvaluationMap.cs:14]: Enum values should automatically handle strings.

File: backend/api/Controllers/PropertyController.cs [hunk 2]
```diff
@@ -80,9 +80,10 @@ public IActionResult GetProperties([FromBody]PropertyFilterModel filter)
             if (!filter.ValidFilter()) throw new BadRequestException("Property filter must contain valid values.");
 
             var properties = new List<PropertyModel>();
-            if (filter.IncludeParcels)
+            var includeBoth = !filter.IncludeParcels && !filter.IncludeBuildings;
+            if (includeBoth || filter.IncludeParcels)
                 properties.AddRange(_mapper.Map<PropertyModel[]>(_pimsService.Parcel.Get((ParcelFilter)filter)));
-            if (filter.IncludeBuildings)
+            if (includeBoth || filter.IncludeBuildings)
```
Comment (Fosol) [inline backend/api/Controllers/PropertyController.cs:86]: Fixed a bug where the default filter wouldn't return anything.

File: backend/api/Helpers/Middleware/LogRequestMiddleware.cs [hunk 3]
```diff
@@ -46,7 +45,7 @@ public LogRequestMiddleware(RequestDelegate next, IWebHostEnvironment env, ILogg
         /// <returns></returns>
         public async Task Invoke(HttpContext context)
         {
-            _logger.LogInformation($"Received HTTP Request {context.Request.Method} user:{context.User.GetDisplayName()} {context.Request.Scheme}://{context.Request.Host}{context.Request.Path}?{context.Request.QueryString}");
+            _logger.LogInformation($"Received HTTP Request {context.Request.Method} user:{context.User.GetDisplayName()} {context.Request.Scheme}://{context.Request.Host}{context.Request.Path}{context.Request.QueryString}");
```
Comment (Fosol) [inline backend/api/Helpers/Middleware/LogRequestMiddleware.cs:48]: Fixed a bug where the log would have a double "??"

File: backend/dal/Services/Admin/Concrete/BuildingService.cs [hunk 4]
```diff
@@ -124,6 +124,19 @@ public override Building Add(Building entity)
         {
             entity.ThrowIfNull(nameof(entity));
 
+            if (entity.Agency != null)
```
Comment (Fosol) [inline backend/dal/Services/Admin/Concrete/BuildingService.cs:127]: This is to ensure we don't attempt to add new related objects when they are attached.  I am not 100% sure this is required, but was running into an issue at one point that this resolved.

File: backend/entities/EvalutationKeys.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+namespace Pims.Dal.Entities
+{
+    /// <summary>
+    /// EvaluationKeys enum, provides a list of valid keys for evaluation values.
+    /// </summary>
+    public enum EvaluationKeys
```
Comment (Fosol) [inline backend/entities/EvalutationKeys.cs:6]: We only currently support two evaluation values.

File: backend/entities/FiscalKeys.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+namespace Pims.Dal.Entities
+{
+    /// <summary>
+    /// FiscalKeys enum, provides a list of valid keys for fiscal values.
+    /// </summary>
+    public enum FiscalKeys
```
Comment (Fosol) [inline backend/entities/FiscalKeys.cs:6]: We only currently support two fiscal values.

File: frontend/src/actionCreators/parcelsActionCreator.spec.tsx [hunk 7]
```diff
@@ -26,10 +26,10 @@ describe('fetchParcels action creator', () => {
     const url = ENVIRONMENT.apiUrl + API.PARCELS(null);
     const mockResponse = { data: { success: true } };
     mockAxios.onGet(url).reply(200, mockResponse);
-    return fetchParcels(null)(dispatch)!.then(() => {
-      expect(requestSpy).toHaveBeenCalledTimes(1);
-      expect(successSpy).toHaveBeenCalledTimes(1);
-      expect(dispatch).toHaveBeenCalledTimes(6);
+    return fetchParcels(null)(dispatch).then(() => {
```
Comment (Fosol) [inline frontend/src/actionCreators/parcelsActionCreator.spec.tsx:29]: Fixed unit test - None of these are called when the argument is `null`.

File: frontend/src/actionCreators/parcelsActionCreator.tsx [hunk 8]
```diff
@@ -14,8 +14,8 @@ export const fetchParcels = (parcelBounds: API.IParcelListParams | null) => (
 ) => {
   if (
     !parcelBounds ||
-    (parcelBounds?.neLatitude !== parcelBounds?.swLatitude &&
-      parcelBounds.neLongitude !== parcelBounds.swLongitude)
+    (parcelBounds?.neLatitude != parcelBounds?.swLatitude &&
+      parcelBounds?.neLongitude != parcelBounds?.swLongitude)
```
Comment (Fosol) [inline frontend/src/actionCreators/parcelsActionCreator.tsx:18]: This may not be required.

File: frontend/src/pages/ManageAccessRequests.tsx [hunk 9]
```diff
@@ -93,7 +93,7 @@ const ManageAccessRequests = () => {
                     }}
                   >
                     {props => (
-                      <Form>
+                      <Form key={accessRequest?.id}>
```
Comment (Fosol) [inline frontend/src/pages/ManageAccessRequests.tsx:96]: Fixed unit-test where it was complaining about not having a unique key.

File: frontend/src/actionCreators/parcelsActionCreator.tsx [hunk 10]
```diff
@@ -14,8 +14,8 @@ export const fetchParcels = (parcelBounds: API.IParcelListParams | null) => (
 ) => {
   if (
     !parcelBounds ||
-    (parcelBounds?.neLatitude !== parcelBounds?.swLatitude &&
-      parcelBounds.neLongitude !== parcelBounds.swLongitude)
+    (parcelBounds?.neLatitude != parcelBounds?.swLatitude &&
```
Comment (devinleighsmith) [inline frontend/src/actionCreators/parcelsActionCreator.tsx:17]: removing the !== will generate a linter error.

**Pull request merged**